dpctl/tensor/libtensor/include/kernels/elementwise_functions/logaddexp.hpp

```diff
@@ -76,12 +73,29 @@ template <typename argT1, typename argT2, typename resT> struct LogAddExpFunctor
 
 #pragma unroll
         for (int i = 0; i < vec_sz; ++i) {
-            resT max = std::max<resT>(in1[i], in2[i]);
-            res[i] = max + std::log1p(std::exp(std::abs(diff[i])));
+            res[i] = impl<resT>(in1[i], in2[i]);
         }
 
         return res;
     }
+
+private:
+    template <typename T> T impl(T const &in1, T const &in2)
+    {
+        T max = std::max<T>(in1, in2);
+        if (std::isnan(max)) {
```

Comment:
the check doesn't work when `in2` has NaN value:
```python
numpy.logaddexp(7, numpy.nan)
# Out: nan

dpt.logaddexp(dpt.asarray(7), dpt.asarray(numpy.nan))
# Out: usm_ndarray(7.69314718)
```
this is because
```cpp
using T = double;
T in1 = 7, in2 = std::numeric_limits<T>::quiet_NaN();

T max = std::max(in1, in2);
// max == 7 && std::isnan(max) == true
```

Reply:
JFI, numpy proposes to have something like:
```cpp
    if (in1 == in2) {
        /* Handles infinities of the same sign without warnings */
        return x + std::log(2);
    }
    else {
        const T tmp = in1 - in2;
        if (tmp > 0) {
            return x + std::log1p(std::exp(-tmp));
        }
        else if (tmp <= 0) {
            return x + std::log1p(std::exp(tmp));
        }
        else {
            /* NaNs */
            return tmp;
        }
    }
```

Reply:
I have pushed changes per feedback from @antonwolfy. Thank you!